fourkitchens/quiltme
# QUILTME-127 PicsArt API service

The service has a method getStyleTransferMediaId(array $options) that accepts an array of options, eg:

```
    $service = \Drupal::service('quiltme_picsart.connector');

    $options = [
      'image_url' => 'https://pyxis.nymag.com/v1/imgs/c49/fa5/d43db44d60dee75b3ed58a3e57ef29d527-Vulturefest5333-1-copy.rvertical.w570.jpg',
      'reference_image_url' => 'https://i.etsystatic.com/16016281/r/il/f57f99/3099413915/il_1588xN.3099413915_qf9h.jpg',
      'level' => 'l5',
      'email' => 'laura@fourkitchens.com',
    ];

    $media_id = $service->getStyleTransferMediaId($options);
```
The above will create a media entity with the returned image from the API.

**Testing steps:**

1. Go to /quiltme
2. Go to https://pr-6-4k-quiltme.pantheonsite.io/admin/content/media
3. You should see an PicsArt-generated media item there 

**Notes:**
I think we should be able to use this in PreSave when the node is created with the user-submitted image to send that image plus a bg image to the API and save the resulting media item to a media reference field on the node.
